### DISCUSSION TITLE
Would REPL get native support?

### DISCUSSION THREAD

**Discussion by purplemint** [carbon-language/carbon-lang#1892] [General]
Python-style REPL is helpful for iterative programming. Understand it is definitely feasible to support using higher level framework, wondering if supporting it natively will be a goal (or non-goal)?

**Comment by OlaFosheimGrostad** [comment]
I don't know what their goals would be, but I guess you could write your own REPL with the reference interpreter? Or are you thinking of something like [cling](https://root.cern/cling/) ?

**Comment by purplemint** [comment]
> I don't know what their goals would be, but I guess you could write your own REPL with the reference interpreter? Or are you thinking of something like [cling](https://root.cern/cling/) ?

Realistically maintaining compatibility with REPL without the support of official goal is difficult, especially given fast iteration in the first few years.

**Comment by geoffromer** [comment]
Native REPL support is not currently a goal. I suspect it won't be possible to provide a really seamless REPL experience without compromising our goal of C++ interoperability, but something like cling might be feasible.

I'm converting this to a discussion, because it seems like more of a question than an issue.

**Reply by purplemint** [reply]
Thanks make sense.
